addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -134,6 +134,12 @@ const Wysiwyg = Widget.extend({
                     if (!$field.hasClass('o_editable_date_field_format_changed')) {
                         $linkedFieldNodes.text($field.data('oe-original-with-format'));
                         $linkedFieldNodes.addClass('o_editable_date_field_format_changed');
+                        const originalDate = $field.data('oe-original');
+                        $(`.oe_hide_on_date_edit[data-oe-original='${originalDate}']`)
+                            .addClass('d-none');
+                        Wysiwyg.setRange(
+                            $(`:not(.oe_hide_on_date_edit)[data-oe-original='${originalDate}']`)[0]
```

Comment:
same here, don't search on whole dom, the element you are looking for is already one of the `$linkedFieldNodes`, find it from there